external-crates/move/crates/move-compiler/src/cfgir/visitor.rs

```diff
@@ -773,37 +773,95 @@ impl<V: SimpleAbsIntConstructor> AbstractInterpreterVisitor for V {
 // utils
 //**************************************************************************************************
 
+pub fn has_special<FCommand, FExp>(
+    cfg: &ImmForwardCFG,
+    mut p_command: FCommand,
+    mut p_exp: FExp,
+) -> bool
+where
+    FCommand: FnMut(&Command) -> bool,
+    FExp: FnMut(&Exp) -> bool,
+{
+    has_special_(cfg, &mut p_command, &mut p_exp)
+}
+
+pub fn has_special_command<FCommand, FExp>(
+    cmd: &Command,
+    mut p_command: FCommand,
+    mut p_exp: FExp,
+) -> bool
+where
+    FCommand: FnMut(&Command) -> bool,
+    FExp: FnMut(&Exp) -> bool,
+{
+    has_special_command_(cmd, &mut p_command, &mut p_exp)
+}
+
+pub fn has_special_exp<F>(e: &Exp, mut p: F) -> bool
+where
+    F: FnMut(&Exp) -> bool,
+{
+    has_special_exp_(e, &mut p)
+}
+
 pub fn calls_special_function(special: &[(&str, &str, &str)], cfg: &ImmForwardCFG) -> bool {
+    has_special(cfg, |_| true, |e| is_special_function(special, e))
+}
+
+pub fn calls_special_function_command(special: &[(&str, &str, &str)], cmd: &Command) -> bool {
+    has_special_command(cmd, |_| true, |e| is_special_function(special, e))
+}
+
+pub fn calls_special_function_exp(special: &[(&str, &str, &str)], e: &Exp) -> bool {
+    has_special_exp(e, |e| is_special_function(special, e))
+}
+
+fn is_special_function(special: &[(&str, &str, &str)], e: &Exp) -> bool {
+    use H::UnannotatedExp_ as E;
+    matches!(
+        &e.exp.value,
+        E::ModuleCall(call) if special.iter().any(|(a, m, f)| call.is(a, m, f)),
+    )
+}
+
+fn has_special_(
+    cfg: &ImmForwardCFG,
+    p_command: &mut impl FnMut(&Command) -> bool,
+    p_exp: &mut impl FnMut(&Exp) -> bool,
+) -> bool {
```

Comment:
Does this work?

```suggestion
fn has_special_<FCmd, FExp>(
    cfg: &ImmForwardCFG,
    p_command: &mut impl FCommand,
    p_exp: &mut impl FExp,
) -> bool 
where
    FCommand: FnMut(&Command) -> bool,
    FExp: FnMut(&Exp) -> bool {
```

Reply:
It did not, but I'll try again to be sure

Reply:
```
error[E0382]: use of moved value: `p_exp`
   --> crates/move-compiler/src/cfgir/visitor.rs:861:86
    |
846 |     mut p_exp: FExp,
    |     --------- move occurs because `p_exp` has type `FExp`, which does not implement the `Copy` trait
...
861 |             C::Mutate(el, er) => has_special_exp_(el, p_exp) || has_special_exp_(er, p_ex...
    |                                                       ----- value moved here         ^^^^^ value used here after move
    |
note: consider changing this parameter type in function `has_special_exp_` to borrow instead if owning the value isn't necessary
   --> crates/move-compiler/src/cfgir/visitor.rs:868:43
    |
868 | fn has_special_exp_<FExp>(e: &Exp, mut p: FExp) -...
    |    ---------------- in this function      ^^^^ this parameter takes ownership of the value
help: if `FExp` implemented `Clone`, you could clone the value
   --> crates/move-compiler/src/cfgir/visitor.rs:843:35
    |
843 | ...mmand, FExp>(
    |           ^^^^ consider constraining this type parameter with `Clone`
...
861 | ...r) => has_special_exp_(el, p_exp) || has_spec...
    |                               ----- you could clone this value
help: consider mutably borrowing `p_exp`
    |
861 |             C::Mutate(el, er) => has_special_exp_(el, &mut p_exp) || has_special_exp_(er, p_exp),
    |                                                       ++++
```